packages/@atjson/offset-annotations/src/annotations/link.ts

```diff
@@ -6,6 +6,7 @@ export class Link extends InlineAnnotation<{
   rel?: string;
   target?: string;
   isAffiliateLink?: boolean;
+  linkStyle?: string;
```

Comment:
What values are we expecting here? Instead of allowing this to be any string can we make this a union type of the allowed values?

Reply:
Why I have given `linkStyle` as string type, that the `Link` annotation gives the freedom to choose any style for the link. With this the link can be styled as "Buy button", "Join now", "Link to own articles" etc... so in the consumer applications can design the link according to the style.

And the another drawback I observed is, when we restrict the link style values with union types, the values what we given which might not suitable for the application, so it might end up with long list of union values.

For your question,
1) Currently, we are expecting one style as "buy-button" ?
2) yes, we can make this field as enum types or union types, 

As union types:
```ts
export class Link extends InlineAnnotation<{
  url: string;
  title?: string;
  rel?: string;
  target?: string;
  isAffiliateLink?: boolean;
  linkStyle?: "buy-button" | "join-now";
}> {
  static vendorPrefix = "offset";
  static type = "link";
}

new Link({
  attributes: {
    url: 'testasdf',
    linkStyle: 'buy-button'
  }
});
```

As enum types:
```ts
export enum LinkDisplayAsTypes {
  'buy-button',
  'join-now',
}

type LinkDisplayAsTypeStrings = keyof typeof LinkDisplayAsTypes;

export class Link extends InlineAnnotation<{
  url: string;
  title?: string;
  rel?: string;
  target?: string;
  isAffiliateLink?: boolean;
  linkStyle?: LinkDisplayAsTypeStrings;
}> {
  static vendorPrefix = "offset";
  static type = "link";
}

new Link({
  attributes: {
    url: 'testasdf',
    linkStyle: 'buy-button'
  }
});
```

Reply:
So suggest me the best option, so I can make the changes accordingly.

Reply:
@bachbui what would be the benefit of enumerating the allowed values? 
@cnrahulrkr I think a value of `buy-button` or `join-now` would be a bit awkward, no? I guess how would you describe these calls to action?

Reply:
Can this just be `button`, ie would it be sufficient to just be able to say that certain links should be styled as a button?
```
export class Link extends InlineAnnotation<{
  url: string;
  title?: string;
  rel?: string;
  target?: string;
  isAffiliateLink?: boolean;
  linkStyle?: "button";
}> {
  static vendorPrefix = "offset";
  static type = "link";
}

```

Reply:
The reason I ask (and @gmedina the reason I think we want to enumerate the values) is that I think it would be best to try to keep a tight rein on the values here, as we want consumers of this format to have some understanding of the content expressed here. This data format `offset-annotations` is meant to be general-purpose and having domain-specific notions of `buy-button` vs `join-now` seems out of place. We could, of course, add these as possible values in some other atjson source that is more domain-specific (such as the Verso source used at Conde Nast).

Regarding the icon button, I wonder if that is not better expressed as a link annotation covering some range that also includes an object annotation representing the icon (that is, the icon is just a part of the content inside the button).

Reply:
- I can keep the value as 'button' only. But, the consumer applications can't differentiate the buttons. @gmedina, for your question asked earlier "what if someone wanted to create a "Join Now" button ?", Its not possible to identify whether its "Join Now" or "Buy now", always the `linkStyle` will be `button` only.

- @bachbui Are you suggesting for creating `Link` annotation or separate annotation for button link in verso source ?
    - If we create `Link` annotation in verso source, there will be two annotation class for Links. I think going forward it will become more complex.
    - If we create separate annotation for `button` in verso source, then this PR is not required and we can continue work with verso source.